openjdk/mmtkHeap.cpp

```diff
@@ -164,7 +164,7 @@ bool MMTkHeap::is_in(const void* p) const {
    //return cp >= committed_low_addr() && cp < committed_high_addr();
 
    //guarantee(false, "is in not supported");
-    return p >= (void*) _start && p < (void*) _end;
+    return p >= (void*) _start && p < (void*) _end || is_in_reserved_malloc(p) == 1;
```

Comment:
For MallocMS this is necessary. But for other GCs, calling into rust code every time at here may slow down the performance.

I'm not familiar with @qinsoon's change but I think we have a global variable in the cpp side to record the selected collector. We can add a branch here and call `is_in_reserved_malloc` only if MallocMS is the selected collector.

Reply:
And I suggest we put an assertion (debug_assert) in `is_in_reserved_malloc` function body to ensure the selected collector is MallocMS. This prevents other collectors from calling this function.

Reply:
There are two basic issues here:
1. I don't think this method was ever called before, because the method was not virtual in the parent class. I guess the reason why this hasn't caused any problems so far is that the implementation of this method is equivalent to its implementation in the parent class.
2. Even if we adding a branch here, there will still be performance penalty, because the branch will look like: `if !matches!(SINGLETON.plan.options().plan, PlanSelector::MarkSweep)`, which is not optimisable at compile time. If we still could use the `if !cfg(feature = "marksweep")` check, there would be no performance overhead.